site/components/form/DateSelector.tsx

```diff
@@ -76,17 +78,36 @@ const DateSelector = <T extends object>({
     optional = false,
     disablePast,
     stateUpdater,
+    reset = false,
+    showError = false,
 }: DateSelectorProps<T>): ReactElement => {
     const [dateValue, setDateValue] = useState<Date | null>(!!disabled || !value ? null : new Date(value));
     const [errors, setErrors] = useState<ErrorInfo[]>(initialErrors);
 
+    useEffect(() => {
+        if (reset) {
+            setDateValue(null);
+        }
+    }, [reset]);
+
     useEffect(() => {
         if (disabled) {
             setErrors([]);
             setDateValue(null);
         }
     }, [disabled]);
 
+    useEffect(() => {
+        if (showError) {
+            setErrors([
+                {
+                    id: inputId,
+                    errorMessage,
+                },
+            ]);
+        }
+    }, [showError, errorMessage, inputId]);
```

Comment:
Similar to the comment above, I don't think any of these dependencies are stateful and will change - so I'm not sure useEffect is appropriate unless I'm misunderstanding something.